bin/s2nd.c

```diff
@@ -546,6 +616,30 @@ int main(int argc, char *const *argv)
         case 'L':
             key_log_path = optarg;
             break;
+        case 'P':
+            input = optarg;
```

Comment:
This is a big switch statement entry. I would suggest moving it to a separate method.